app/policies/wiki_policy.rb

```diff
@@ -1,6 +1,35 @@
 class WikiPolicy < ApplicationPolicy
 
   def index?
-    true
+    public_or_visible
+  end
+
+  # Stop user from seeing a private wiki by typing in private wiki url.
+  def show?
+    public_or_visible
+  end
+
```

Comment:
*If* we were being exhaustive, we'd also need to add methods for `destroy?` and `update?` in `WikiPolicy`. Those methods provided in `ApplicationPolicy` aren't adequate for handling collaborators.

However, I think it more important that we move on to the next project. Implementing `destroy?` and `update?` in here can wait until a later date.